pkg/deps/summary.go

```diff
@@ -0,0 +1,114 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package deps
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"regexp"
+	"text/template"
+
+	"github.com/apache/skywalking-eyes/pkg/header"
+	"github.com/apache/skywalking-eyes/pkg/license"
+)
+
+var fileNamePattern = regexp.MustCompile(`[^a-zA-Z0-9\\.\-]`)
+
+type SummaryRenderContext struct {
+	LicenseContent string                       // Current project license content
+	Groups         []*SummaryRenderLicenseGroup // All dependency license groups
+}
+
+type SummaryRenderLicenseGroup struct {
+	Name string                  // Aggregate all same license ID dependencies
+	Deps []*SummaryRenderLicense // Same license ID dependencies
+}
+
+type SummaryRenderLicense struct {
+	Name      string // Dependency name
+	Version   string // Dependency version
+	Location  string // The filename of generated license file
+	LicenseID string // License ID
+}
+
+func GenerateDependencyLicenseFilename(result *Result) string {
```

Comment:
This method is not used in `--summary` flag, it's used in `--output` flag, why put it here (`summary.go`)?

Reply:
Because I provide the `Location` field [into the template](https://github.com/apache/skywalking-eyes/pull/103/files#diff-d235c0f816b8e113b0fe5f573372f3bb09ed71a44c553056e452c5a9b5e59dc5R101), It could help for getting generated license file path.